projects/plugins/jetpack/modules/sso.php

```diff
@@ -192,6 +205,83 @@ public function invite_user_to_wpcom() {
 		wp_die();
 	}
 
+	/**
+	 * Handles logic to revoke user invite.
+	 */
+	public function revoke_user_invite_to_wpcom() {
+		check_admin_referer( 'jetpack-sso-revoke-user-invite', 'revoke_invite_nonce' );
+		$nonce = wp_create_nonce( 'jetpack-sso-invite-user' );
+
+		if ( ! current_user_can( 'promote_users' ) ) {
+			$query_params = array(
+				'jetpack-sso-invite-user'  => 'failed',
+				'jetpack-sso-invite-error' => 'invalid-revoke-permissions',
+				'_wpnonce'                 => $nonce,
+			);
+
+			return self::create_error_notice_and_redirect( $query_params );
+		} elseif ( isset( $_GET['user_id'] ) ) {
+			$user_id = intval( wp_unslash( $_GET['user_id'] ) );
+			$user    = get_user_by( 'id', $user_id );
+
+			if ( ! $user ) {
+
+				$query_params = array(
+					'jetpack-sso-invite-user'  => 'failed',
+					'jetpack-sso-invite-error' => 'invalid-user-revoke',
+					'_wpnonce'                 => $nonce,
+				);
+
+				return self::create_error_notice_and_redirect( $query_params );
+			}
+
+			$has_pending_invite = self::has_pending_wpcom_invite( $user_id );
```

Comment:
Let's use the ID I mention above, this request is expensive. 

```suggestion
			$has_pending_invite = $_GET['invite_id'];
```